### PR TITLE
opencl2-header rename to opencl-header, also update dependent packages

### PR DESCRIPTION
**Pull request by aryalaadi**
Renaming opencl2-header to opencl-header because its misleading since this ships headers for OpenCL 3.0 specs as well. Unless the 2 in opencl2-header is supposed to signify something else, its better to rename it. Also updated every dependent packages.
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/opencl-headers/template [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Template file for 'opencl2-headers'
-pkgname=opencl2-headers
+# Template file for 'opencl-headers'
+pkgname=opencl-headers
```
Comment (ahesford) [inline srcpkgs/opencl-headers/template:2]: This should be `OpenCL-Headers` if we're going to rename the package, and there should be a meta-package called `opencl2-headers` that pulls this in for backwards compatibility.

File: srcpkgs/opencl2-headers/template
Comment (aryalaadi) [inline srcpkgs/opencl2-headers/template:1]: @ahesford is this what a metapackage is? im not sure

**Review by ahesford — CHANGES_REQUESTED** [review]
The transitional metapackage should be a subpackage of the one that replaces it. See, for example, the `stow` template, which was renamed from `perl-stow`.

Also, don't strip my name from the maintainer field.

**Comment by aryalaadi** [comment]
looks good now? also i assumed this package was abandoned or smth my bad

File: srcpkgs/OpenCL-Headers/template [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+revision=1
+build_style=cmake
+short_desc="OpenCL 3.0.14 (Open Computing Language) header files"
+maintainer="Andrew J. Hesford <ajh@sideband.org>"
+license="Apache-2.0"
+homepage="https://github.com/KhronosGroup/OpenCL-Headers"
+distfiles="https://github.com/KhronosGroup/OpenCL-Headers/archive/v${version}.tar.gz"
+checksum=0ce992f4167f958f68a37918dec6325be18f848dee29a4521c633aae3304915d
+provides="OpenCL-Headers-${version}_${revision}"
+replaces="OpenCL-Headers>=0"
```
Comment (ahesford) [inline srcpkgs/OpenCL-Headers/template:13]:
```suggestion
provides="opencl-headers-${version}_${revision}"
replaces="opencl-headers>=0"
```
These lines replace an old, removed `opencl-headers` package, so they need to be carried forward from the `opencl2-headers` template verbatim.

**Review by ahesford — CHANGES_REQUESTED** [review]
You'll need to make a symlink from `srcpkgs/opencl2-headers` to `OpenCL-Headers` so that the transitional metapackage is built.

**Comment by ahesford** [comment]
Thanks. I cleaned up the commits and split the dependant changes, including revbumping `ocl-icd` so that the dependency change to `ocl-icd-devel` is reflected in the package.

**Pull request closed without merging**